File: src/app/model/appointment.ts
import { Staff } from "./auth";
import { Branch } from "./branch";

export interface Appointment {
    appointment_id: string;
    member_id: string;
    staff_id: string;
    appointment_date: Date;
    start_time: string;
    end_time: string;
    purpose: string;
    notes: string | null;
    status: string;
    branch_id: string;
    created_date: Date;
    modified_date: Date;
    staff: Staff;
    branch: Branch;
  }

File: LibraryManagement.Core/Entities/Appointment.cs
using System;
using System.Collections.Generic;

namespace LibraryManagement.Core.Entities;

public partial class Appointment
{
    public string AppointmentId { get; set; } = null!;

    public string MemberId { get; set; } = null!;

    public string StaffId { get; set; } = null!;

    public DateOnly AppointmentDate { get; set; }

    public TimeOnly StartTime { get; set; }

    public TimeOnly EndTime { get; set; }

    public string Purpose { get; set; } = null!;

    public string? Notes { get; set; }

    public string? Status { get; set; }

    public string? BranchId { get; set; }

    public DateTime CreatedDate { get; set; }

    public DateTime? ModifiedDate { get; set; }

    public virtual LibraryBranch? Branch { get; set; }

    public virtual Member Member { get; set; } = null!;

    public virtual Staff Staff { get; set; } = null!;
}
